src/extension.ts

```diff
@@ -34,31 +34,51 @@ export function activate(context: vscode.ExtensionContext) {
 
     context.subscriptions.push(disposable1);
 
-    const disposable2 = vscode.commands.registerCommand('ucl-rsdg.validateAtf', async () => {
-        const editor = vscode.window.activeTextEditor;
-        const fileName = basename(editor.document.uri.fsPath);
-        const fileContent = editor.document.getText();
-        let fileProject: string;
-        try {
-            fileProject = getProjectCode(fileContent);
-        } catch (err) {
-            vscode.window.showErrorMessage(`Could not validate: ${err}`);
-            return;
-        }
-        // The validate function is currently not mapped to the appropriate logging functions
-        const result = await validate(fileName,fileProject,fileContent);
-        handleResult(result, editor);
+    const disposable2 = vscode.commands.registerCommand('ucl-rsdg.validateAtf', () => {
+        workWithServer("validate", validate);
         });
 
     context.subscriptions.push(disposable2);
 
+    context.subscriptions.push(
+        vscode.commands.registerCommand('ucl-rsdg.lemmatiseAtf', () => {
+            workWithServer("lemmatise", lemmatise)
+        })
+    );
+
     context.subscriptions.push(
         vscode.commands.registerCommand('ucl-rsdg.arabicPreview', () => {
             PreviewPanel.createOrShow(context.extensionUri);
         })
     );
 }
 
+/**
+ * Base function for performing server-related tasks with the editor contents.
+ *  
+ * Will first get some basic information from the editor (filename, project
+ * code, text contents), then call another function to communicate with
+ * the server, and finally display the results of that communication.
+ *
+ * @param verb a textual description of the action, for reporting purposes
+ * @param callback the function to call after getting the editor information
+ */
+async function workWithServer(verb: string, callback: ServerFunction): Promise<void> {
```

Comment:
Oh, still haven't understood promises properly! They are supposed to handle asynchronous requests, right? 

Should we call verb "command" instead? That's what Steve calls those I think.

Reply:
I thought about "command", but I wanted it to be clearer that it's an actual verb that goes into the success/error messages.